src/ATen/templates/TensorDerived.cpp

```diff
@@ -20,17 +20,22 @@ const char * ${Tensor}::toString() const {
 }
 
 IntList ${Tensor}::sizes() const {
-  return IntList(reinterpret_cast<int64_t*>(tensor->size),dim());
+  int64_t d = ${THTensor_nDimension};
```

Comment:
nit: in strides() this was tensor->nDimension

Reply:
nit: I had to read this twice to figure out why it didn't need a ScalarCheck. Maybe not it is reading the underlying THTensor dim which for dim() == 0 will be 1. Otherwise it looks like this cannot return {}

Reply:
For the first comment, I think we should change strides; dim() already uses ${THTensor_nDimension}` so this is consistent with that usage.

In the second comment I'm having trouble parsing this sentence: "Maybe not it is reading the underlying THTensor dim which for dim() == 0 will be 1", could you clarify?  If the question is why this can return {} for scalars, that didn't change for this PR, but I'll add a comment to make it more clear.

Reply:
err, actually on the first point, ${THTensor_nDimension}` is not defined when we are substituting in TensorDense (where strides() is defined), so leaving this as is is probably the cleanest.

Reply:
that is what happens when I rush :), yep just wanted a comment.